### PR TITLE
add list_events use case

### PR DESCRIPTION
**Pull request by dhemasnurjaya**

### CODE REVIEW

**Comment by da1nerd** [comment]
Task linked: [CU-868atkwfk Continually pull from connected devices](https://app.clickup.com/t/868atkwfk)

File: lib/src/feature/data/repositories/event_repository_impl.dart [hunk 1]
```diff
@@ -83,6 +83,7 @@ class EventRepositoryImpl extends EventRepository {
     List<EventModel> events;
 
     try {
+      // TODO: (opendoor) need to expose this method in this repository
```
Comment (da1nerd) [inline lib/src/feature/data/repositories/event_repository_impl.dart:86]: Don't we already have access to the data source in opendoor?

**Review by da1nerd — APPROVED** [review]
The code looks fine, just wondering if it's necessary.

**Comment by da1nerd** [comment]
Oh, if we expose this, will we technically not need to manually read from the datasource? That would be nice.

**Comment by dhemasnurjaya** [comment]
> The code looks fine, just wondering if it's necessary.

story behind this:
event_sink stores generated events (from `add` method) to it's local cache. So when a device need to read it's events for push and pull (in Open Door's nearby connections context) it needs to be able to read that stored events.

before we move event_sink remote adapter out, that local cache doesn't need to be exposed because it's only used internally between a remote server and a device.

**Pull request merged**